tee-worker/litentry/core/assertion-build/src/nodereal/weirdo_ghost_gang_holder.rs

```diff
@@ -0,0 +1,163 @@
+// Copyright 2020-2023 Trust Computing GmbH.
+// This file is part of Litentry.
+//
+// Litentry is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+#[cfg(all(feature = "std", feature = "sgx"))]
+compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the same time");
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+extern crate sgx_tstd as std;
+
+use core::result;
+
+use lc_credentials::nodereal::weirdo_ghost_gang_holder::WeirdoGhostGangHolderAssertionUpdate;
+use lc_data_providers::nodereal_jsonrpc::{
+	GetTokenBalance721Param, NftApiList, NoderealChain, NoderealJsonrpcClient,
+};
+
+use crate::*;
+use lc_data_providers::Error as DataProviderError;
+
+const NFT_TOKEN_ADDRESS: &str = "0x9401518f4EBBA857BAA879D9f76E1Cc8b31ed197";
+
+fn check_has_nft(
+	client: &mut NoderealJsonrpcClient,
+	address: &str,
+) -> result::Result<bool, DataProviderError> {
+	let param = GetTokenBalance721Param {
+		token_address: NFT_TOKEN_ADDRESS.into(),
+		account_address: address.into(),
+		block_number: "latest".into(),
+	};
+
+	match client.get_token_balance_721(&param) {
+		Ok(res) => {
+			debug!("Get token balance 721 response: {:?}", res);
+			Ok(res > 0)
+		},
+		Err(e) => {
+			error!("Error get token balance 721 by param: {:?}, {:?}", param, e);
+			Err(e)
+		},
+	}
+}
+
+pub fn build(req: &AssertionBuildRequest) -> Result<Credential> {
+	debug!("WeirdoGhostGang holder");
+
+	let mut has_nft = false;
+	let mut client = NoderealJsonrpcClient::new(NoderealChain::Eth);
```

Comment:
Then why do we need to have `NoderealChain` enums? Is it rather preparation for the upcoming requirements?

Reply:
Yeah preparation for expending the vc to different network, e.g.  NoderealChain::Bsc etc later